src/celmodel/model.cpp

```diff
@@ -9,15 +9,125 @@
 // of the License, or (at your option) any later version.
 
 #include <algorithm>
-#include <cassert>
-#include <functional>
 
-#include <celutil/reshandle.h>
+#include <Eigen/Geometry>
+
 #include "model.h"
 
-using namespace cmod;
-using namespace Eigen;
-using namespace std;
+
+namespace cmod
+{
+namespace
+{
+bool
+operator<(const Material::Color& c0, const Material::Color& c1)
+{
+    if (c0.red() < c1.red())
```

Comment:
can be replaced with std::tie

Reply:
`std::tie` doesn't work here because the results of the functions are temporaries. `std::forward_as_tuple` compiles but I'm not completely sure whether this ends up producing dangling references, so I've refactored this as a friend operator bool in the material files.